### PR TITLE
shell: fix open state toggling of the dropdowns on the shell

### LINKED ISSUE #17944: Auto close menu
Can you make menu automatically close/hide when clicked or tapped anywhere on the screen. I am talking about Session, Help, Host and System menu. It is more user friendly and shouldn't be much coding. Specially on mobile device it is a bit annoying.

### PR DESCRIPTION
**Pull request by KKoukiou**
Explanation of the commit message in inline comment. For more info on this HACK please read https://github.com/wellyshen/react-cool-onclickoutside/blob/master/README.md#detecting-iframe-clicks

Fixes #17944

### CODE REVIEW

**Review by jelly — DISMISSED** [review]
TIL!

I'd have never thought about our iframes causing this!

File: pkg/shell/topnav.jsx [hunk 1]
```diff
@@ -64,6 +64,20 @@ export class TopNav extends React.Component {
...
+    componentDidMount() {
+        /* This is a HACK for catching lost clicks on the pages which live in iframes so as to close dropdown menus on the shell.
+         * Note: Clicks on an <iframe> element won't trigger document.documentElement listeners, because it's literally different page with different security domain.
+         * However, when clicking on an iframe moves focus to its content's window that triggers the main window.blur event.
+         */
+        window.addEventListener("blur", this.handleClickOutside);
+    }
+
+    componentWillUnmount() {
+        window.removeEventListener("blur", this.handleClickOutside);
```
Comment (cockpituous) [inline pkg/shell/topnav.jsx:80]: These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17959-20221128-125239-e85efe43-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Review by garrett — CHANGES_REQUESTED** [review]
This is working as I would expect for the help and session  menus.

However, it's not working for the host and system menus in mobile mode.

While this is already a great improvement, would it be possible to add support for the other menus too? Thanks!

(Unrelated: I also see there's a dark theme problem with the host switcher in mobile now. So when you're checking mobile, don't be surprised.)

**Comment by KKoukiou** [comment]
@garrett fixed the system selector (although it's actually not much relevant as it takes most of the space in mobile) but the hosts one I won't fix in this commit because the code for hosts expander code is mixed in for mobile and desktop. I am sure we only want to affect the mobile one as you said, and that will require a quite big rewrite of the code in order that it's achievable.

File: pkg/shell/topnav.jsx (hunk 1 above)
Comment (cockpituous) [inline pkg/shell/topnav.jsx:80]: These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17959-20221129-115044-2afa12d5-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Review by jelly — APPROVED** [review]

**Comment by garrett** [comment]
@KKoukiou: I had already fixed the host selector in the dark theme PR; I was mentioning it here just to not be surprised about it being broken in this PR. :grin: We'll have a fix for it landing soon, one way or another.

**Pull request merged**